# Ejemplo de Seber (1984) usando James test

<img src="https://raw.githubusercontent.com/fhernanb/fhernanb.github.io/master/my_docs/logo_unal_color.png" alt="drawing" width="200"/>

En este ejemplo se desea estudiar el siguiente conjunto de hipótesis usando un nivel de significancia de 0.07.

$H_0: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2$

$H_A: \boldsymbol{\mu}_1 \neq \boldsymbol{\mu}_2$

Asumiendo que las matrices $\boldsymbol{\Sigma}_1$ y $\boldsymbol{\Sigma}_2$ son <font color="red">desconocidas y diferentes</font>.

Las matrices muestrales son

\begin{equation}
\bar{\mathbf{X}}_1 = 
\begin{pmatrix}
9.82 \\
15.06
\end{pmatrix}
\end{equation}

\begin{equation}
\bar{\mathbf{X}}_2= 
\begin{pmatrix}
13.05 \\
22.57
\end{pmatrix}
\end{equation}

\begin{equation}
\mathbf{S}_1 = 
\begin{pmatrix}
120 & -16.3 \\
-16.3 & 17.8
\end{pmatrix}
\end{equation}

\begin{equation}
\mathbf{S}_2 = 
\begin{pmatrix}
81.8 & 32.1 \\
32.1 & 53.8
\end{pmatrix}
\end{equation}

con $n_1=16$ y $n_2=11$ sujetos por cada grupo.

<img src="ph_james.png" alt="" width="600"/>

In [1]:
import numpy as np
import scipy.stats

A continuación los datos del ejemplo

In [2]:
xbarra_1 = np.matrix([[9.82],
                      [15.06]])

xbarra_2 = np.matrix([[13.05],
                      [22.57]])

S1 = np.matrix([[120, -16.3],
                [-16.3, 17.8]])

S2 = np.matrix([[81.8, 32.1],
                [32.1, 53.8]])

n1 = 16
n2 = 11

p = 2   # número de variables

Primero vamos a calcular la matriz combinada $\tilde{\mathbf{S}}$.

In [3]:
S1_tilde = S1 / n1
S2_tilde = S2 / n2
S_tilde = S1_tilde + S2_tilde
print(S_tilde)

[[14.93636364  1.89943182]
 [ 1.89943182  6.00340909]]


Ahora vamos a calcular $A$.

In [4]:
mat_aux_1 = np.matmul(np.linalg.inv(S_tilde), S1_tilde)
mat_aux_2 = np.matmul(np.linalg.inv(S_tilde), S2_tilde)

aux1 = np.trace(mat_aux_1)
aux2 = np.trace(mat_aux_2)

A = 1 + (aux1 ** 2 / (n1-1) + aux2 ** 2 / (n2-1)) / (2*p)
print(A)

1.0480216500545976


Ahora vamos a calcular $B$.

In [5]:
aux1 = np.trace(np.matmul(mat_aux_1, mat_aux_1)) / (n1-1) + np.trace(np.matmul(mat_aux_2, mat_aux_2)) / (n2-1)
aux2 =np.trace(mat_aux_1) ** 2 / (n1-1) + np.trace(mat_aux_2) ** 2 / (n2-1)

B = (aux1 + aux2 / 2) / (p*(p+2))
print(B)

0.02650981376101031


Ahora calculamos $\delta$.

In [6]:
delta = A + B * scipy.stats.chi2.ppf(q=1-0.07, df=p)
print(delta)

1.1890146266969683


Ahora calcularmos $T^2$.

In [7]:
aux = xbarra_1 - xbarra_2
T2 = np.matmul(np.matmul(aux.T, np.linalg.inv(S_tilde)), aux)
T2.item()  # Para obtener el T2 como un escalar y no como una matriz

9.445541402394888

Para calcular el valor crítico con un $\alpha=0.07$ hacemos lo siguiente

In [8]:
scipy.stats.chi2.ppf(q=1-0.07, df=p) * delta  # Valor crítico

6.323798160207584

<div class="alert alert-success">
  <strong>Resultado</strong> Como el estadístico es $T^2=9.45$, que es mayor que el valor crítico $6.32$, entonces hay evidencias suficientes para rechazar $H_0: \boldsymbol{\mu}_1 = \boldsymbol{\mu}_2$.
</div>